File: app/src/main/java/com/example/test/BinDao.java
package com.example.test;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;
import io.reactivex.rxjava3.core.Single;

@Dao
public interface BinDao {

    @Query("SELECT * FROM historyBin2 ORDER BY timestamp DESC")
    Single<List<CardEntity>> getHistoryCardBin();

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertBin(CardEntity cardEntity);
}

File: app/src/main/java/com/example/test/data/database/BinDao.kt
package com.example.test.data.database

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import com.example.test.data.model.CardEntity
import io.reactivex.rxjava3.core.Single

@Dao
interface BinDao {

    @androidx.room.Query("SELECT * FROM historyBin2 ORDER BY timestamp DESC")
    fun getHistoryCardBin(): Single<List<CardEntity>>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insertBin(cardEntity: CardEntity)
}